bot/exts/events/hacktoberfest/_utils.py

```diff
@@ -0,0 +1,452 @@
+
+    For each PR:
+    {
+        "repo_url": str
+        "repo_shortname": str (e.g. "python-discord/sir-lancebot")
+        "created_at": datetime.datetime
+        "number": int
+    }
+
+    Otherwise, return empty list.
+    None will be returned when the GitHub user was not found.
+    """
+    log.info(f"Fetching Hacktoberfest Stats for GitHub user: '{github_username}'")
+    base_url = "https://api.github.com/search/issues"
+
+    hacktoberfest_timeframe = f"{CURRENT_YEAR} - 09 - 30T10: 00Z..{CURRENT_YEAR} - 11 - 01T12: 00Z"
+    query_params = (
+        f"+type:pr"                               # Only get PR if it's:
+        f"+is:public"                             # - public
+        f"+author:{quote_plus(github_username)}"  # - by the user's github username
+        f"+-is:draft"                             # - not a draft
+        f"+created:{hacktoberfest_timeframe}"     # - made within hacktoberfest.
+        f"&per_page=100"  # Limit results per-page returned from API (100 is the maximum)
+    )
+
+    log.debug(f"GitHub query parameters generated: {query_params}")
+
+    # The `params` argument needs to be specified as a string to stop aiohttp percent-encoding
+    jsonresp = await fetch_url(bot, base_url, STATS_REQUEST_HEADERS, f"q={query_params}")
+    if "message" in jsonresp:
+        # One of the parameters is invalid, short circuit for now
+        api_message = jsonresp["errors"][0]["message"]
+
+        # Ignore logging non-existent users or users we do not have permission to see
+        if api_message == GITHUB_NONEXISTENT_USER_MESSAGE:
+            log.debug(f"No GitHub user found named '{github_username}'")
+            return
+        else:
+            log.error(f"GitHub API request for '{github_username}' failed with message: {api_message}")
+        return []  # No October PRs were found due to error
+
+    if jsonresp["total_count"] == 0:
+        # Short circuit if there aren't any PRs
+        log.info(f"No October PRs found for GitHub user: '{github_username}'")
+        return []
+
+    logging.info(f"Found {len(jsonresp['items'])} Hacktoberfest PRs for GitHub user: '{github_username}'")
+    outlist = []  # List of pr information dicts that will get returned
+    hackto_topics = {}  # Cache whether each repo has the appropriate topic (bool values)
+    for item in jsonresp["items"]:
+        shortname = get_shortname(item["repository_url"])
+        itemdict = {
+            "repo_url": f"https://www.github.com/{shortname}",
+            "repo_shortname": shortname,
+            "created_at": datetime.strptime(
+                item["created_at"], "%Y-%m-%dT%H:%M:%SZ"
+            ),
+            "number": item["number"]
+        }
+
+        # If the PR has 'invalid' or 'spam' labels, the PR must be
+        # either merged or approved for it to be included
+        if has_label(item, ["invalid", "spam"]):
+            if not await is_accepted(bot, itemdict):
+                continue
```

Comment:
You can merge the nested if conditions

```suggestion
        if has_label(item, ["invalid", "spam"]) and not await is_accepted(
            bot, itemdict
        ):
            continue
```